From c4c6a8873cd2a0aa818f0e2a08c9da84ac05a358 Mon Sep 17 00:00:00 2001
From: Ankit Agrawal <ankit.agarwal@intel.com>
Date: Thu, 18 May 2023 15:22:18 +0530
Subject: [PATCH] Removed Verity Support.

Removing drm-verity support as it no longer
supported by google.

Tracked-On: OAM-106853
Signed-off-by: Tanuj Tekriwal <tanuj.tekriwal@intel.com>
---
 Android.mk | 76 +++---------------------------------------------------
 1 file changed, 3 insertions(+), 73 deletions(-)

diff --git a/Android.mk b/Android.mk
index 5e72383..28e9b16 100644
--- a/Android.mk
+++ b/Android.mk
@@ -155,40 +155,6 @@ LOCAL_MODULE := kernelflinger-$(TARGET_BUILD_VARIANT)
 
 
 # if dm-verity is disabled for eng purpose skip the oem-cert
-ifeq ($(PRODUCT_SUPPORTS_VERITY), true)
-kf_intermediates := $(call intermediates-dir-for,EFI,kernelflinger)
-
-VERITY_CERT := $(kf_intermediates)/verity.cer
-PADDED_VERITY_CERT := $(kf_intermediates)/verity.padded.cer
-OEMCERT_OBJ := $(kf_intermediates)/oemcert.o
-
-$(VERITY_CERT): $(PRODUCTS.$(INTERNAL_PRODUCT).PRODUCT_VERITY_SIGNING_KEY).x509.pem
-	$(transform-pem-cert-to-der-cert)
-
-$(PADDED_VERITY_CERT): $(VERITY_CERT)
-	$(call pad-binary, 4096)
-
-ifeq ($(TARGET_UEFI_ARCH),x86_64)
-    ELF_OUTPUT := elf64-x86-64
-else
-    ELF_OUTPUT := elf32-i386
-endif
-
-sym_binary := $(shell echo _binary_$(PADDED_VERITY_CERT) | sed "s/[\/\.-]/_/g")
-$(OEMCERT_OBJ): $(PADDED_VERITY_CERT)
-	mkdir -p $(@D) && \
-	$(EFI_OBJCOPY) --input binary --output $(ELF_OUTPUT) --binary-architecture i386 $< $@ && \
-	$(EFI_OBJCOPY) --redefine-sym $(sym_binary)_start=_binary_oemcert_start \
-                       --redefine-sym $(sym_binary)_end=_binary_oemcert_end \
-                       --redefine-sym $(sym_binary)_size=_binary_oemcert_size \
-                       --rename-section .data=.oemkeys $@ $@
-
-LOCAL_GENERATED_SOURCES := $(OEMCERT_OBJ)
-else # PRODUCT_SUPPORTS_VERITY
-ifneq (,$(filter user userdebug, $(TARGET_BUILD_VARIANT)))
-
-endif
-endif # PRODUCT_SUPPORTS_VERITY
 
 kf_intermediates := $(call intermediates-dir-for,EFI,kernelflingeravb)
 
@@ -202,7 +168,7 @@ BOOTLOADER_AVB_KEY_PATH := $(BOARD_AVB_KEY_PATH)
 endif
 
 $(AVB_PK): $(BOOTLOADER_AVB_KEY_PATH)
-	external/avb/avbtool extract_public_key --key $< --output $@
+	external/avb/avbtool.py extract_public_key --key $< --output $@
 
 $(PADDED_AVB_PK): $(AVB_PK)
 	$(call pad-binary, 4096)
@@ -255,9 +221,6 @@ endif
 
 LOCAL_CFLAGS := $(SHARED_CFLAGS)
 
-ifeq ($(PRODUCT_SUPPORTS_VERITY), true)
-LOCAL_OBJCOPY_FLAGS := -j .oemkeys
-endif
 
 LOCAL_OBJCOPY_FLAGS := -j .oemkeys
 
@@ -306,7 +269,7 @@ BOOTLOADER_AVB_KEY_PATH := $(BOARD_AVB_KEY_PATH)
 endif
 
 $(KFINS_AVB_PK): $(BOOTLOADER_AVB_KEY_PATH)
-	external/avb/avbtool extract_public_key --key $< --output $@
+	external/avb/avbtool.py extract_public_key --key $< --output $@
 
 $(KFINS_PADDED_AVB_PK): $(KFINS_AVB_PK)
 	$(call pad-binary, 4096)
@@ -397,36 +360,6 @@ ifneq ($(strip $(KERNELFLINGER_USE_UI)),false)
     LOCAL_SRC_FILES += \
         ux.c
 endif
-ifeq ($(PRODUCT_SUPPORTS_VERITY),true)
-keys4abl_intermediates := $(call intermediates-dir-for,ABL,keys)
-
-ABL_VERITY_CERT := $(keys4abl_intermediates)/verity.cer
-ABL_PADDED_VERITY_CERT := $(keys4abl_intermediates)/verity.padded.cer
-ABL_OEMCERT_OBJ := $(keys4abl_intermediates)/oemcert.o
-
-$(ABL_VERITY_CERT): $(PRODUCTS.$(INTERNAL_PRODUCT).PRODUCT_VERITY_SIGNING_KEY).x509.pem
-	$(transform-pem-cert-to-der-cert)
-
-$(ABL_PADDED_VERITY_CERT): $(ABL_VERITY_CERT)
-	$(call pad-binary, 4096)
-
-ifeq ($(TARGET_IAFW_ARCH),x86_64)
-    ELF_OUTPUT := elf64-x86-64
-else
-    ELF_OUTPUT := elf32-i386
-endif
-
-abl_sym_binary := $(shell echo _binary_$(ABL_PADDED_VERITY_CERT) | sed "s/[\/\.-]/_/g")
-$(ABL_OEMCERT_OBJ): $(ABL_PADDED_VERITY_CERT)
-	mkdir -p $(@D) && \
-	$(EFI_OBJCOPY) --input binary --output $(ELF_OUTPUT) --binary-architecture i386 $< $@ && \
-	$(EFI_OBJCOPY) --redefine-sym $(abl_sym_binary)_start=_binary_oemcert_start \
-                       --redefine-sym $(abl_sym_binary)_end=_binary_oemcert_end \
-                       --redefine-sym $(abl_sym_binary)_size=_binary_oemcert_size \
-                       --rename-section .data=.oemkeys $@ $@
-
-LOCAL_GENERATED_SOURCES := $(ABL_OEMCERT_OBJ)
-endif #.PRODUCT_SUPPORTS_VERITY == true
 
 keys4abl_intermediates := $(call intermediates-dir-for,ABL,keys4abl)
 
@@ -440,7 +373,7 @@ BOOTLOADER_AVB_KEY_PATH := $(BOARD_AVB_KEY_PATH)
 endif
 
 $(ABL_AVB_PK): $(BOOTLOADER_AVB_KEY_PATH)
-	external/avb/avbtool extract_public_key --key $< --output $@
+	external/avb/avbtool.py extract_public_key --key $< --output $@
 
 $(ABL_PADDED_AVB_PK): $(ABL_AVB_PK)
 	$(call pad-binary, 4096)
@@ -506,9 +439,6 @@ ifneq ($(strip $(KERNELFLINGER_USE_UI)),false)
         ux.c
 endif
 
-ifeq ($(PRODUCT_SUPPORTS_VERITY),true)
-LOCAL_GENERATED_SOURCES := $(ABL_OEMCERT_OBJ)
-endif
 
 LOCAL_GENERATED_SOURCES += $(ABL_AVB_PK_OBJ)
 LOCAL_C_INCLUDES := \
-- 
2.17.1

